src/loki/search.cc

```diff
@@ -642,16 +650,20 @@ struct bin_handler_t {
           // the last one wasnt in the radius so replace it with this one because its better or is
           // in the radius
           if (!last_in_radius) {
+            if (closer_external_reachable)
+              p_itr->closest_external_reachable = batch->back().sq_distance;
```

Comment:
as we are rejecting reachable candidates we keep track of how close the closest one was